EventFilter/L1TRawToDigi/plugins/OmtfUnpacker.cc

```diff
@@ -0,0 +1,876 @@
+  
+          for (unsigned int iframe=1; iframe <=3; iframe++) {
+  
+            uint16_t frame = (iframe==1) ?  data.frame1() : ( (iframe==2) ?  data.frame2() : data.frame3() );
+            if (frame==0) continue;
+            rpcrawtodigi::RecordCD recordCD(frame);
+            records.add(recordCD);
+  
+            LogTrace("") <<"OMTF->RPC Event isComplete: "<<records.complete() <<records.print(rpcrawtodigi::DataRecord::StartOfBXData); // << std::endl; 
+            LogTrace("") <<"OMTF->RPC Event:             "<<records.print(rpcrawtodigi::DataRecord::StartOfTbLinkInputNumberData) << std::endl; 
+            LogTrace("") <<"OMTF->RPC Event:             "<<records.print(rpcrawtodigi::DataRecord::ChamberData)
+                      <<" lb:"<< recordCD.lbInLink() 
+                      <<" part: "<< recordCD.partitionNumber() 
+                      <<" partData: "<<recordCD.partitionData() 
+                      << std::endl << std::endl;
+  
+            if (records.complete()) formater.recordUnpack( records,  producedRPCDigis.get(), 0,0);
+          }
+        }
+
+
+
+        //
+        // CSC data
+        //
+        if (DataWord64::csc==recordType) {
+          CscDataWord64   data(*word);
+          OmtfEleIndex omtfEle(fedHeader.sourceID(), bh.getAMCNumber()/2+1, data.linkNum());
+          std::map<OmtfEleIndex,CSCDetId>::const_iterator icsc = omtf2csc_.find(omtfEle);
+          if (icsc==omtf2csc_.end()) {LogTrace(" ") <<" CANNOT FIND key: " << omtfEle << std::endl; continue; }
+          CSCDetId cscId = omtf2csc_[omtfEle];
+          LogTrace("") <<"OMTF->CSC "<<cscId << std::endl; 
+          LogTrace("") << data << std::endl;
+          CSCCorrelatedLCTDigi digi(data.hitNum(), //trknmb
+                                    data.valid(), 
+                                    data.quality(),
+                                    data.wireGroup(),
+                                    data.halfStrip(),
+                                    data.clctPattern(),
+                                    data.bend(),
+                                    data.bxNum()+3) ;
+          LogTrace("") << digi << std::endl;
+          producedCscLctDigis->insertDigi( cscId, digi); 
+
+        } 
+
+        //
+        // OMTF (muon) data
+        //
+        if (DataWord64::omtf==recordType) {
+          OmtfDataWord64   data(*word);
+          LogTrace("") <<"OMTF->MUON " << std::endl;
+          LogTrace("") << data << std::endl;
+          l1t::tftype  overlap = (fedId==1380) ? l1t::tftype::omtf_neg :  l1t::tftype::omtf_pos;
+          unsigned int iProcessor = bh.getAMCNumber()/2;   //0-5 
+          l1t::RegionalMuonCand digi;
+          digi.setHwPt(data.pT());
+          digi.setHwEta(data.eta());
+          digi.setHwPhi(data.phi());
+          digi.setHwSign(data.ch());
+          digi.setHwSignValid(data.vch());
+          digi.setHwQual(data.quality());
+          std::map<int, int> trackAddr;
```

Comment:
So in this case it would make sense to also unpack these bits on the uGMT unpacker side. Already from a comparison between OMTF and uGMT unpackers point of view this makes sense.